MSVC debug runtime iterators don't want to be decremented past begin()

--- hunspell/src/hunspell/affixmgr.cxx.orig	2016-04-15 23:30:37.555875079 +0200
+++ hunspell/src/hunspell/affixmgr.cxx	2016-04-15 23:30:43.669875027 +0200
@@ -4638,7 +4638,7 @@
       return;
 
   int neg = 0;
-  for (std::string::iterator k = piece.begin() + piece.size() - 1; k >= piece.begin(); --k) {
+  for (std::string::iterator k = piece.begin() + piece.size() - 1; ; --k) {
     switch (*k) {
       case '[': {
         if (neg)
@@ -4666,6 +4666,8 @@
           *(k + 1) = *k;
       }
     }
+    if (k == piece.begin())
+        break;
   }
 }
 
